### DISCUSSION TITLE
How we can scan LLM  based application?

### DISCUSSION THREAD

**Discussion by neeraj-mahapatra** [NVIDIA/garak#431] [Q&A]
As I worked through Garak, I was able to conduct probes in models such as the GPT-2 and GPT-3.5-turbo. However, how can I scan the entire Gen AI app?

**Comment by leondz** [comment]
Hi,

This depends on the API that the model is exposed via. We have a lot of different `generators` to help with this - see e.g. https://reference.garak.ai/en/latest/generators.html. If there's something you'd like to use that's not on the list, the `RestGenerator` is pretty flexible and can work with most REST API formats.
